examples/counter/scripts/abi-generator.ts

```diff
@@ -0,0 +1,117 @@
+// Clarinet extention 101
+//
+// This extension is introspecting the contracts of the project it's running from, 
+// and produce a Typescript data structure autocomplete friendly, that developers
+// can use in their frontend code.
+//
+// When running:
+// $ clarinet run --allow-write scripts/abi-generator.rs
+//
+// This script will write a file at the path artifacts/contracts.ts:
+//
+// export namespace CounterContract {
+//     export const ADDRESS = "ST000000000000000000002AMW42H";
+//     export const NAME = "counter";
+//     export enum Public {
+//         INCREMENT = "increment",
+//         DECREMENT = "decrement",
+//     }   
+//     export enum ReadOnly {
+//         READ = "read",
+//     }
+// }
+//
+// By importing this file in their frontend code, developers can use constants, instead 
+// of hard coding principals and strings:
+// 
+// await makeContractCall({
+//     contractAddress: CounterContract.ADDRESS,
+//     contractName: CounterContract.NAME,
+//     functionName: CounterContract.Public.INCREMENT,
+//     functionArgs: [CounterContract.Args.INCREMENT(namespace)], // or so
```

Comment:
Is this `functionArgs` example a TODO / WIP?

Reply:
Addressed with latest